alf/networks/actor_networks.py

```diff
@@ -140,18 +140,20 @@ def forward(self, observation, state=()):
         actions = []
         i = 0
         for layer, spec in zip(self._action_layers, self._flat_action_spec):
+            layer.summary_name = (
+                'summarize_output/' + self.name + '.action_layer.' + str(i))
             action = layer(encoded_obs)
+            layer.summary_name = ""
             action = spec_utils.scale_to_spec(action, spec)
 
             if alf.summary.should_summarize_output():
-                a_name = ('summarize_output/' + self.name + '.action_layer.' +
-                          str(i) + '.action.output_norm')
-                if not self.training:
-                    a_name += ".eval"
-                a = action
+                a_name = (
```

Comment:
should_summarize_output() will always return True if config.summarize_output is True. But summary is only performed when should_record_summaries() return True. Need to avoid the overhead by conditioning on should_record_summaries() too.